### PR TITLE
hm2

### PR DESCRIPTION
**Pull request by ArazureX**
new file:   Files/raw/sales/2022-08-09/sales_2022-08-09_1.json
	new file:   Files/raw/sales/2022-08-09/sales_2022-08-09_2.json
	new file:   Files/raw/sales/2022-08-09/sales_2022-08-09_3.json
	new file:   Files/raw/sales/2022-08-09/sales_2022-08-09_4.json
	new file:   Files/stg/sales/2022-08-09/sales_2022-08-09_1.avro
	new file:   Files/stg/sales/2022-08-09/sales_2022-08-09_2.avro
	new file:   Files/stg/sales/2022-08-09/sales_2022-08-09_3.avro
	new file:   Files/stg/sales/2022-08-09/sales_2022-08-09_4.avro
	new file:   Job_1/main.py
	new file:   Job_2/job_2_main.py
	new file:   requirements.txt

### CODE REVIEW

File: Job_1/main.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+from flask import Flask, jsonify, request
+import requests
+import os
+import json
+import shutil
+
+app = Flask(__name__)
```
Comment (mikolaz27) [inline Job_1/main.py:7]: split this big function into small, use the single responsibility principle

File: Job_1/main.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+AUTH_TOKEN = os.environ['AUTH_TOKEN']
```
Comment (mikolaz27) [inline Job_1/main.py:8]: better to move it to separate module settings

File: Job_1/main.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@app.route('/fetch_api_data', methods=['POST'])
+def fetch_api_data():
+
+    
+    api_url = ('https://fake-api-vycpfa6oca-uc.a.run.app/sales')
+
+   
+    headers = {'Authorization':AUTH_TOKEN}
```
Comment (mikolaz27) [inline Job_1/main.py:17]: better to move it to separate module settings

File: Job_1/main.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    
+   
+    for filename in os.listdir(raw_dir):
+        file_path = os.path.join(raw_dir, filename)
+        try:
+            if os.path.isfile(file_path) or os.path.islink(file_path):
+                os.unlink(file_path)
+            elif os.path.isdir(file_path):
+                shutil.rmtree(file_path)
+        except Exception as e:
```
Comment (mikolaz27) [inline Job_1/main.py:34]: catch specific error not just exception

File: Job_2/job_2_main.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    },
+    {
+      "name": "product",
+      "type": "string"
+    },
+    {
+      "name": "price",
+      "type": "float"
+    }
+  ]
```
Comment (mikolaz27) [inline Job_2/job_2_main.py:34]: better to move schema to separate file

File: Job_2/job_2_main.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            # Переконайтеся, що json_data_list є списком словників
+            for json_data in json_data_list:
+                # Конвертація поля purchase_date
+                purchase_date_str = json_data['purchase_date']
+                purchase_date_obj = datetime.strptime(purchase_date_str, '%Y-%m-%d')
+                purchase_date_int = (purchase_date_obj - datetime(1970, 1, 1)).days
+                json_data['purchase_date'] = purchase_date_int
+
+                # Запишіть дані у файл AVRO
+                with open(avro_file_path, 'wb') as af:
```
Comment (mikolaz27) [inline Job_2/job_2_main.py:69]: the same here, split by smaller functions

File: Job_2/job_2_main.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    
+    # Перелічіть всі файли JSON у вхідній директорії
+    json_files = [f for f in os.listdir(raw_dir) if f.endswith('.json')]
+
+    for json_file in json_files:
+        # Створіть повні шляхи до файлів
+        json_file_path = os.path.join(raw_dir, json_file)
+        avro_file_path = os.path.join(stg_dir, json_file.replace('.json', '.avro'))
+
+        # Прочитайте файл JSON
```
Comment (mikolaz27) [inline Job_2/job_2_main.py:56]: use english comments

**Review by ArazureX — COMMENTED** [review]
55

**Pull request merged**